stepper_eip_driver/src/stepper_node.cpp

```diff
@@ -111,6 +116,13 @@ int main(int argc, char *argv[])
       // Collect status from controller, convert to ROS message format.
       stepper.updateDriveStatus(stepper.getDriveData());
 
+      joint_state.header.stamp = ros::Time::now();
+      joint_state.name.resize(1);
+      joint_state.position.resize(1);
+      joint_state.name[0] = "stepper_pos";
+      joint_state.position[0] = stepper.ss.current_position / 1000.0;
```

Comment:
The [JointState ](http://docs.ros.org/melodic/api/sensor_msgs/html/msg/JointState.html)message specifies units of meters for prismatic axes. Subscriber ROS nodes will always expect distances to be in meters so it makes sense to perform the conversion here.

Reply:
So that becomes a problem if the device configured units are not mm ... I think.

Reply:
@schornakj wrote:
>  so it makes sense to perform the conversion here.

even stronger: if you're publishing `JointState` msgs, you are *required* to do the conversion, as otherwise you're violating the contract that's embedded in that message type.